packages/core/src/extensions/Blocks/nodes/ListItemBlock.ts

```diff
@@ -0,0 +1,246 @@
+              state.selection.anchor === state.selection.head;
+
+            if (selectionAtBlockStart && selectionEmpty) {
+              return commands.BNSetContentType(
+                state.selection.from,
+                "textBlock"
+              );
+            }
+
+            return false;
+          }),
+        // Regular backspace behaviour.
+        () => commands.deleteSelection(),
+      ]);
+
+    const handleEnter = () =>
+      this.editor.commands.first(({ commands }) => [
+        () =>
+          // Changes list item block to a text block if both the content and selection are empty.
+          commands.command(({ state }) => {
+            const { node, contentType } = getBlockInfoFromPos(
+              state.doc,
+              state.selection.from
+            )!;
+
+            if (contentType.name !== "listItemBlock") return false;
+
+            const selectionEmpty =
+              state.selection.anchor === state.selection.head;
+
+            if (selectionEmpty && node.textContent.length === 0) {
+              return commands.BNSetContentType(
+                state.selection.from,
+                "textBlock"
+              );
+            }
+
+            return false;
+          }),
+
+        () =>
+          // Splits the current block, moving content inside that's after the cursor to a new block of the same type
+          // below.
+          commands.command(({ state, chain }) => {
+            const { node, contentType } = getBlockInfoFromPos(
+              state.doc,
+              state.selection.from
+            )!;
+
+            if (contentType.name !== "listItemBlock") return false;
+
+            if (node.textContent.length > 0) {
+              chain()
+                .deleteSelection()
+                .BNSplitBlock(state.selection.from, true)
+                .run();
+
+              return true;
+            }
+
+            return false;
+          }),
+      ]);
+
+    return {
+      Backspace: handleBackspace,
+      Enter: handleEnter,
+    };
+  },
+
+  parseHTML() {
+    return [
+      {
+        tag: "li",
+        getAttrs: (element) => {
+          if (typeof element === "string") {
+            return false;
+          }
+
+          const parent = element.parentElement;
+
+          if (parent === null) {
+            return false;
+          }
+
+          // Gets type of list item (ordered/unordered) based on parent element's tag ("ol"/"ul").
+          if (parent.tagName === "UL") {
+            return { type: "unordered" };
+          }
+
+          if (parent.tagName === "OL") {
+            return { type: "ordered" };
+          }
+
+          return false;
+        },
+      },
+    ];
+  },
+
+  renderHTML() {
+    return [
+      "div",
+      {
+        "data-node-type": "block-content",
+        "data-content-type": this.name,
+        class: styles.blockContent,
+      },
+      ["li", 0],
+    ];
+  },
+});
+
+// ProseMirror Plugin which automatically assigns
```

Comment:
Yeah probably a good idea. I'm still not sure about the file structure since text and headings only need 1 file while lists also need a plugin which should probably go in a separate file. So it seems a bit clunky to have a separate directory for each block type since some will only have 1 file, but I'm also not a fan of having both nodes and plugins defined in the same directory.

Reply:
Let's create a separate directory per block type for now